sahi/model.py

```diff
@@ -663,3 +664,167 @@ def _create_object_prediction_list_from_original_predictions(
+            prediction_result = self.model([image])
+
+        self._original_predictions = prediction_result
+
+    @property
+    def num_categories(self):
+        """
+        Returns number of categories
+        """
+        return len(self.category_mapping)
+
+    @property
+    def has_mask(self):
+        """
+        Returns if model output contains segmentation mask
+        """
+        has_mask = self.model.with_mask
+        return has_mask
+
+    @property
+    def category_names(self):
+        return self.category_mapping
+
+    def _create_object_prediction_list_from_original_predictions(
+        self,
+        shift_amount_list: Optional[List[List[int]]] = [[0, 0]],
+        full_shape_list: Optional[List[List[int]]] = None,
+    ):
+        """
+        self._original_predictions is converted to a list of prediction.ObjectPrediction and set to
+        self._object_prediction_list_per_image.
+        Args:
+            shift_amount_list: list of list
+                To shift the box and mask predictions from sliced image to full sized image, should
+                be in the form of List[[shift_x, shift_y],[shift_x, shift_y],...]
+            full_shape_list: list of list
+                Size of the full image after shifting, should be in the form of
+                List[[height, width],[height, width],...]
+        """
+        original_predictions = self._original_predictions
+        category_mapping = self.category_mapping
+
+        # compatilibty for sahi v0.8.15
+        if isinstance(shift_amount_list[0], int):
+            shift_amount_list = [shift_amount_list]
+        if full_shape_list is not None and isinstance(full_shape_list[0], int):
+            full_shape_list = [full_shape_list]
+
+        # parse boxes, masks, scores, category_ids from predictions
+        from sahi.utils.torchvision import COCO_CLASSES
+
+        prediction_class = [COCO_CLASSES[i] for i in list(original_predictions[0]["labels"].numpy())]
+        prediction_boxes = [
+            [(i[0], i[1]), (i[2], i[3])] for i in list(original_predictions[0]["boxes"].detach().numpy())
+        ]
+        prediction_score = list(original_predictions[0]["scores"].detach().numpy())
+        prediction_thresh = [prediction_score.index(x) for x in prediction_score if x > self.confidence_threshold][-1]
+
+        boxes = prediction_boxes[: prediction_thresh + 1]
+        score = list(original_predictions[0]["scores"].detach().numpy())
+        category_name = prediction_class[: prediction_thresh + 1]
+
+        category_map = {}
+        for i in range(len(COCO_CLASSES)):
+            category_map[COCO_CLASSES[i]] = i
+            category_map[i] = COCO_CLASSES[i]
+        category_id = [category_map[i] for i in category_name]
+
+        try:
+            masks = (self._original_predictions[0]["masks"] > 0.5).squeeze().detach().cpu().numpy()
+            masks = masks[: self._original_predictions + 1]
+        except:
+            masks = None
```

Comment:
Yakalanan exception belirsiz, kod okunabilirliğini düşüren bir kısım, başka bir şekilde replace etmek daha iyi olabilir, (if-else veya spesifik exception'ı yakalyarak). Burada neden execution'ın except bloğuna gireceği çok belli değil dolayısıyla kodun davranışı direkt anlaşılmıyor.

Reply:
Ekstra bir fikir olarak, yukarıda önerdiğim spesifik exception catch'in daha iyi olduğunu düşünmekle birlikte, alternatif olarak except bloğuna comment yazılabilir neden except'e ihtiyaç olduğuna dair.

Reply:
burada buyuk ihtimalle mask outputu olmayan modeller handle edilmis, dedigin gibi bir comment bile yetebilir :+1: